planning/costmap_generator/test/test_points_to_costmap.cpp

```diff
@@ -0,0 +1,225 @@
+// Copyright 2022 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <costmap_generator/points_to_costmap.hpp>
+
+#include <gtest/gtest.h>
+using pointcloud = pcl::PointCloud<pcl::PointXYZ>;
+class PointsToCostmapTest : public ::testing::Test
+{
+protected:
+  void SetUp() override { rclcpp::init(0, nullptr); }
+
+  ~PointsToCostmapTest() override { rclcpp::shutdown(); }
+
+  grid_map::GridMap construct_gridmap();
+
+public:
+  double grid_resolution_ = 1;
+  double grid_length_x_ = 20;
+  double grid_length_y_ = 20;
+  double grid_position_x_ = 0;
+  double grid_position_y_ = 0;
+};
+
+grid_map::GridMap PointsToCostmapTest::construct_gridmap()
+{
+  grid_map::GridMap gm;
+
+  // set gridmap size
+  gm.setFrameId("map");
+  gm.setGeometry(
+    grid_map::Length(grid_length_x_, grid_length_y_), grid_resolution_,
+    grid_map::Position(grid_position_x_, grid_position_y_));
+
+  // set initial value
+  gm.add("points", 0);
+
+  // set car postion in map frame to center of grid
+  grid_map::Position p;
+  p.x() = 0;
+  p.y() = 0;
+  gm.setPosition(p);
+
+  return gm;
+}
+
+// grid_y
+// |  map_x-------
+// |            |
+// |            |
+// |            |
+// |            map_y
+// |__________________grid_x
+TEST_F(PointsToCostmapTest, TestMakeCostmapFromPoints_validPoints)
+{
+  // construct pointcloud in map frame
+  pointcloud in_sensor_points;
+
+  in_sensor_points.width = 3;
+  in_sensor_points.height = 1;
+  in_sensor_points.is_dense = false;
+  in_sensor_points.resize(in_sensor_points.width * in_sensor_points.height);
+
+  in_sensor_points.points[0].x = 0.7;
+  in_sensor_points.points[0].y = 1;
+  in_sensor_points.points[0].z = 1;
+
+  in_sensor_points.points[1].x = 1.1;
+  in_sensor_points.points[1].y = 1;
+  in_sensor_points.points[1].z = 4;
+
+  in_sensor_points.points[2].x = 1.4;
+  in_sensor_points.points[2].y = 2;
+  in_sensor_points.points[2].z = 2.7;
+
+  grid_map::GridMap gridmap = construct_gridmap();
+
+  PointsToCostmap point2costmap;
+  double maximum_height_thres = 5;
+  double minimum_lidar_height_thres = 0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
+
+  std::string gridmap_layer_name = "points";
```

Comment:
Some other parts can be `const`.

```suggestion
  const std::string gridmap_layer_name = "points";
```